src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -323,7 +324,22 @@ InterpreterSelectQuery::InterpreterSelectQuery(
         table_lock = storage->lockForShare(context->getInitialQueryId(), context->getSettingsRef().lock_acquire_timeout);
         table_id = storage->getStorageID();
         if (!metadata_snapshot)
-            metadata_snapshot = storage->getInMemoryMetadataPtr();
+        {
+            if (storage->getName() != "Distributed")
+            {
+                metadata_snapshot = storage->getInMemoryMetadataPtr();
+            }
+            else
+            {
+                const StorageDistributed * storage_distributed = static_cast<const StorageDistributed *>(storage.get());
+                StoragePtr storage_replicated = DatabaseCatalog::instance().getTable(
```

Comment:
It only works if table is remote table is present in local shard.

Reply:
I afraid yes. Do you have any suggestion how to make it work when remote table is not present in local shard? Thanks a lot.

Reply:
I have fixed the failed stateless tests by switching to previous behavior in some cases.